matchzoo/engine/base_model.py

```diff
@@ -189,6 +189,43 @@ def evaluate(
         return self._backend.evaluate(x=x, y=y,
                                       batch_size=batch_size, verbose=verbose)
 
+    def evaluate_with_metrics(
+            self,
+            generator: 'engine.BaseGenerator',
+            metrics: typing.List[str],
+            batch_size: int = 128
+    ) -> typing.Union[float, typing.List[float]]:
+        """
+        Evaluate the model with metrics on.
+
+        See :meth:`keras.models.Model.evaluate` for more details.
+
+        :param generator: A generator, an instance of
+            :class:`engine.BaseGenerator`.
+        :param metrics: metrics used on evaluation.
+        :param batch_size: number of samples per gradient update
+        :return: list of scalars (losses as well as metrics). The
+            attribute `model.backend.metrics_names` will give you
+            the display labels for the scalar outputs.
+
+        """
+        qinfo = {}
+        for (x, y) in generator:
```

Comment:
for batch in generator? I think `generator[0]` is the first batch, then you can access x and y through batch.